src/Cache/TemplateCache.php

```diff
@@ -25,14 +26,40 @@ class TemplateCache
      */
     protected ModuleHookLegacyRepository $hookModuleRepository;
 
+    /*
+     * @var Configuration
+     */
+    protected Configuration $configuration;
+
+    protected const CACHE_KEY = 'IS_MAINMENU_REQUEST_CACHE_KEY';
+
     public function __construct(
         \Is_mainmenu $module,
         \Context $context,
-        ModuleHookLegacyRepository $hookModuleRepository
+        ModuleHookLegacyRepository $hookModuleRepository,
+        Configuration $configuration
     ) {
         $this->module = $module;
         $this->context = $context;
         $this->hookModuleRepository = $hookModuleRepository;
+        $this->configuration = $configuration;
+    }
+
+    public function getContextBasedRequestCacheKey(): string
+    {
+        return md5($this->getRequestCacheKey() . '|' . $this->context->language->id . '|' . $this->context->shop->id . '|' . $this->context->customer->id_default_group);
+    }
+
+    private function getRequestCacheKey(): int
```

Comment:
Shouldn't that be a string as in `getContextBasedRequestCacheKey`?

```suggestion
    private function getRequestCacheKey(): string
```

Reply:
now that I've written that, I'm not sure anymore what was your idea for that ;-)

Reply:
@kpodemski it's actually cache key for get request for front end ajax request. It's still work in progress. This cache key will be incremented very time any changes to menu will occur.